### DISCUSSION TITLE
Page tree navigation inheritance

### DISCUSSION THREAD

**Discussion by bbcakes4real** [WikiTransformationProject/wikitraccs-releases#17] [General]
Hi!

Does this tool inherit the page tree structure from Confluence? For example:

![image](https://user-images.githubusercontent.com/123505469/214394879-10ededdb-29c8-498f-8c93-99c39592b465.png)

Would it add the transformed pages to the SharePoint site like this?

![image](https://user-images.githubusercontent.com/123505469/214395097-659ae762-e313-4132-9813-10707e26b9d8.png)

or, do the pages have to be added to SharePoint navigation manually?

![image](https://user-images.githubusercontent.com/123505469/214395285-c073e1f3-e6b7-44ee-af1a-c128702f392f.png)

Thanks so much!

**Comment by heinrich-ulbricht** [comment]
**EDIT: There now is a page tree solution available in Microsoft AppSource. See my [comment below](https://github.com/WikiTransformationProject/wikitraccs-releases/discussions/17#discussioncomment-5883213).**

---

@bbcakes4real All migrated pages are created in the Site Pages library of (configurable) target SharePoint sites. There is no native hierarchy in SharePoint that could display a deep hierarchy. The out of the box navigations cannot be nested deep enough to be of use. And they would have to be updated manually when the structure changes.

So currently there is no out-of-the-box tree view for migrated Confluence pages. The metadata is there though in SharePoint: page IDs, parent page IDs etc. - all is there to display a dynamic hierarchy of pages in SharePoint. A SharePoint Framework Solution could be made to display such a hierarchy.

What's your take on this? Have you ideas how and where to display deeply nested structures from Confluence in the SharePoint modern experience?

(Sorry for not getting back to you earlier, notifications were broken :( )

**Comment by RobinAtRestore** [comment]
For us, this is a serious blocker to migration as we have thousands of pages all neatly organised in Confluence.  when you export a site from confluence manually, you seem to get an index.html file which looks to be an HTMLised structure - is there anything similar when the tool migrates a space ?

**Comment by heinrich-ulbricht** [comment]
@RobinAtRestore I hear you and you are not alone! That's why I created WikiPakk - a Confluence breadcrumb and page tree, available via Microsoft AppSource ("SharePoint Store") 👉 https://appsource.microsoft.com/en-us/product/office/wa200005439

The tree uses hierarchy information that is already migrated by WikiTraccs. And it works with new SharePoint pages as well.

Please check it out. I'd be glad to get feedback if that solves your use cases.
